apps/SageAccountWeb/src/components/TwoFactorAuth/ResetTwoFactorAuth.test.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import { ResetTwoFactorAuth } from './ResetTwoFactorAuth'
+import { render, screen, waitFor, within } from '@testing-library/react'
+import TestWrapper, { TestWrapperProps } from '../../tests/TestWrapper'
+import { RESET_2FA_SIGNED_TOKEN_PARAM } from '../../Constants'
+import { TwoFactorAuthResetToken } from '@sage-bionetworks/synapse-types'
+import { vitest } from 'vitest'
+import { userEvent } from '@testing-library/user-event'
+import { server } from '../../mocks/node.js'
+import { oauth2PromptFor2FAHandler } from '../../mocks/handlers/oauth2Handlers'
+
+describe('ResetTwoFactorAuth', () => {
+  beforeAll(() => {
+    server.listen()
+  })
+  beforeEach(() => {
+    vitest.clearAllMocks()
+  })
+
+  afterEach(() => {
+    server.resetHandlers()
+  })
+
+  afterAll(() => {
+    server.close()
+  })
+
+  const mockTwoFactorAuthResetToken: TwoFactorAuthResetToken = {
+    concreteType: 'org.sagebionetworks.repo.model.auth.TwoFactorAuthResetToken',
+    userId: 123,
+    hmac: 'foo',
+    version: 1,
+    expiresOn: new Date().toISOString(),
+    createdOn: new Date().toISOString(),
+  }
+  const hexEncodedResetToken = Buffer.from(
+    JSON.stringify(mockTwoFactorAuthResetToken),
+    'utf-8',
+  ).toString('hex')
+
+  function renderComponent(wrapperProps?: TestWrapperProps) {
+    const user = userEvent.setup()
+    render(<ResetTwoFactorAuth />, {
+      wrapper: props => <TestWrapper {...props} {...wrapperProps} />,
+    })
+    return { user }
+  }
+
+  it('Shows an error if no twoFAResetToken is present', async () => {
+    renderComponent()
+
+    const alert = await screen.findByRole('alert')
+    within(alert).getByText('No token was found in the URL.')
+  })
+  it('Handles resetting 2FA with password', async () => {
+    const { user } = renderComponent({
+      memoryRouterProps: {
+        initialEntries: [
+          `?${RESET_2FA_SIGNED_TOKEN_PARAM}=${hexEncodedResetToken}`,
+        ],
+      },
+    })
+
+    await waitFor(() => {
+      expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+    })
+
+    const loginWithPasswordButton = screen.getByRole('button', {
+      name: 'Sign in with your email',
+    })
+
+    await user.click(loginWithPasswordButton)
+
+    const passwordField = screen.getByLabelText('Password')
+    const disableTwoFactorAuthButton = screen.getByRole('button', {
+      name: 'Disable Two-Factor Authentication',
+    })
+
+    expect(disableTwoFactorAuthButton).toBeDisabled()
+
+    await user.type(passwordField, 'myPassword')
+
+    expect(disableTwoFactorAuthButton).toBeEnabled()
+
+    await user.click(disableTwoFactorAuthButton)
+
+    await screen.findByText(
+      '2FA has been successfully disabled on your account.',
+    )
+  })
+
+  it.todo(
```

Comment:
This is tricky to get to work, we may need to render the entire app instead of just the component, since the SSO behavior depends on rerouting the user and this component knows nothing about routes.